app/exporters/bangladesh_dhis2_exporter.rb

```diff
@@ -0,0 +1,34 @@
+class BangladeshDhis2Exporter
+  require "dhis2"
+
+  def self.export
+    periods = (current_month_period.advance(months: -24)..current_month_period)
+
+    exporter = Dhis2Exporter.new(
+      facility_identifiers: FacilityBusinessIdentifier.dhis2_org_unit_id,
+      periods: periods,
+      data_elements_map: CountryConfig.current.fetch(:dhis2_data_elements)
+    )
+
+    exporter.export do |facility_identifier, period|
+      repository = Reports::Repository.new(facility_identifier.facility.region, periods: periods)
```

Comment:
We should instantiate the repository outside the export block. Currently a new object would be created every time we call the block and we will not benefit from memoization that happens within repository.